controller/transfer/project_transfer_manager.py

```diff
@@ -287,11 +288,162 @@ def import_file(
+                embedding_item["model"] = model
+                embedding_name_mapping[embedding_item.get("name")] = name
+                embedding_item["name"] = name
+            
+            attribute_id = embedding_item.get("attribute_id")
+            embedding_name = embedding_item.get("name")
+            if attribute_id:
+                attribute_id = attribute_ids_by_old_id.get(attribute_id)
+            else:
+                attribute_name = __get_attribute_name_from_embedding_name(
+                    embedding_name
+                )
+                attribute_id = attribute_ids_by_old_name[attribute_name]
+
+            finished_at_str = "finished_at" in embedding_item
+            if not finished_at_str:
+                embedding_item["finished_at"] = sql.func.now()
+
+            embedding_object = embedding.create(
+                project_id=project_id,
+                attribute_id=attribute_id,
+                name=embedding_name,
+                state="FINISHED",
+                created_by=import_user_id,
+                custom=embedding_item.get(
+                    "custom",
+                ),
+                type=embedding_item.get(
+                    "type",
+                ),
+                started_at=embedding_item.get(
+                    "started_at",
+                ),
+                finished_at=embedding_item.get(
+                    "finished_at",
+                ),
+                platform=embedding_item.get(
+                    "platform",
+                ),
+                model=embedding_item.get(
+                    "model",
+                ),    
+            )
+            embedding_ids[
+                embedding_item.get(
+                    "id",
+                )
+            ] = embedding_object.id
+
+        if data.get(
+            "embedding_tensors_data",
+        ): 
+            for embedding_tensor_item in data.get(
+                "embedding_tensors_data",
+            ):
+                embedding.create_tensor(
+                    project_id=project_id,
+                    record_id=record_ids.get(
+                        embedding_tensor_item.get(
+                            "record_id",
+                        )
+                    ),
+                    embedding_id=embedding_ids.get(
+                        embedding_tensor_item.get(
+                            "embedding_id",
+                        )
+                    ),
+                    data=embedding_tensor_item.get(
+                        "data",
+                    ),
+                )
+
+    def __replace_embedding_name(soure_code: str, embedding_name_mapping: Dict[str, str]) -> str:
+        for embedding_name in embedding_name_mapping.keys():
+            if embedding_name in soure_code:
+                code =  soure_code.replace(
+                    embedding_name,
+                    embedding_name_mapping[embedding_name])
```

Comment:
i dont think this is enough. since an embedding name could be partially included so e.g. 

embedding_name = "ABC"

embedding_name = "ABCD"

the direct replace would replace both which can result in errors. I first thought regex word bounds might do the trick but a dash (-) is also a word boundarie so I'd run it twice with
"<embedding_name>" 
'<embedding_name>' 

that way we should be relatively sure that we don't replace something we don't want to.

Reply:
How exactly could a embedding name be partially included? 

the old ones are build like this


attribute_name-embedding_type-model_name

e.g.

headline-classification-distilbert-

Maybe I am not thinking hard enough, but what example would there be where one embedding name is included in another one?

Reply:
Now that I think about it maybe the attribute part could be problematic e.g.

headline-classification-roberta-base

custom-headline-classification-roberta-base

--> will change it

Reply:
changed it